### PR TITLE
ncm-spma: yumng: Major improvements

### PR DESCRIPTION
**Pull request by jnovy**
- run all operations in one single transaction
- use saner repository consistency check and display
  less criptic message once YUM repos are broken
- always refresh YUM metadata
- use less verbose and more readable output
- warn in case transaction (not) installed/removed
  expected packages
- avoid breaking transaction by protected packages
  (mainly systemd)

### CODE REVIEW

**Comment by stdweird** [comment]
test this please

File: ncm-spma/src/main/perl/spma/yumng.pm [hunk 1]
```diff
@@ -12,17 +12,14 @@ use warnings;
 use NCM::Component;
 our $EC  = LC::Exception::Context->new->will_store_all;
 our @ISA = qw(NCM::Component);
-use EDG::WP4::CCM::Path 16.8.0 qw(unescape);
+use EDG::WP4::CCM::Element qw(unescape);
```
Comment (stdweird) [inline ncm-spma/src/main/perl/spma/yumng.pm]: please revert

File: ncm-spma/src/main/perl/spma/yumng.pm [hunk 2]
```diff
@@ -635,169 +568,42 @@ sub Configure
...
-                $rpm_noepoch =~ s/^.*://;
-                if ( index($rpm_noepoch, $white_pkg) == 0 || match_glob($white_pkg, $rpm_noepoch) ) {
-                    $will_remove->delete($rpm);
-                    $whitelisted->insert($rpm);
-                }
-            }
-        }
+    # End here in case of --noaction.
+    if ($NoAction) {
+        return 1;
```
Comment (stdweird) [inline ncm-spma/src/main/perl/spma/yumng.pm]: add a info message of some sort

File: ncm-spma/src/main/perl/spma/yumng.pm (hunk 1 above)
Comment (jnovy) [reply ncm-spma/src/main/perl/spma/yumng.pm]: It would help to describe why ;-)

File: ncm-spma/src/main/perl/spma/yumng.pm (hunk 2 above)
Comment (jnovy) [reply ncm-spma/src/main/perl/spma/yumng.pm]: The message is already there:

[INFO] Dry run, no changes will be performed (--noaction flag set)

Comment (jnovy) [reply ncm-spma/src/main/perl/spma/yumng.pm]: and:

[INFO] running component: spma
---------------------------------------------------------
[INFO] would mark state of component as needing to run
[INFO] note: running component spma in noaction mode

in the component directly. Assuming we don't need to add this message for the third time in the log.

**Comment by jnovy** [comment]
FYI: this code runs in our env for several months en-masse now. It's well tested. Feel free to report any problems should you encounter some.

**Comment by stdweird** [comment]
@jnovy can you also cleanup the description? it still has the template text

File: ncm-spma/src/main/perl/spma/yumng.pm (hunk 1 above)
Comment (stdweird) [reply ncm-spma/src/main/perl/spma/yumng.pm]: https://github.com/quattor/CCM/issues/168

File: ncm-spma/src/main/perl/spma/yumng.pm [hunk 3]
```diff
@@ -180,12 +114,12 @@ sub Configure
 
     my $repos = $config->getElement(REPOS_TREE)->getTree();
     my $t     = $config->getElement(CMP_TREE)->getTree();
-
+
```
Comment (jrha) [inline ncm-spma/src/main/perl/spma/yumng.pm]: Can you strip this white-space please.

File: ncm-spma/src/main/perl/spma/yumng.pm [hunk 4]
```diff
@@ -180,12 +114,12 @@ sub Configure
...
     my $repos = $config->getElement(REPOS_TREE)->getTree();
     my $t     = $config->getElement(CMP_TREE)->getTree();
-
+    
     # Display system info
     if ( defined($t->{quattor_os_release}) ) {
         $self->info("target OS build: ", $t->{quattor_os_release});
     }
-
+
```
Comment (jrha) [inline ncm-spma/src/main/perl/spma/yumng.pm]: Can you strip this white-space please.

**Comment by jnovy** [comment]
@stdweird - Ok, I will clean up the description (I will catch up with you via email to avoid churn here)

@jrha - I will, thanks for noticing

**Comment by jnovy** [comment]
Seems automatic tests got stuck at some point?

**Comment by jrha** [comment]
@hpcugentbot test this please

**Pull request merged**